zerver/tests/test_sessions.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import
+
+from zerver.lib.sessions import (
+    user_sessions,
+    delete_session,
+    delete_user_sessions,
+    delete_realm_user_sessions,
+    delete_all_user_sessions,
+    delete_all_deactivated_user_sessions,
+)
+
+from zerver.models import (
+    UserProfile, get_user_profile_by_id,
+    get_user_profile_by_email, get_realm
+)
+
+from zerver.lib.test_classes import ZulipTestCase
+
+
+class TestSessions(ZulipTestCase):
+
+    def do_test_session(self, user, action, expected_result):
+        self.login(user)
+        self.assertIn('_auth_user_id', self.client.session)
+        action()
+        if expected_result:
+            result = self.client_get('/')
+            self.assertEqual('/login', result.url)
+        else:
+            self.assertIn('_auth_user_id', self.client.session)
+
+    def test_delete_session(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        for session in user_sessions(user_profile):
+            delete_session(session)
+        result = self.client_get("/")
+        self.assertEqual('/login', result.url)
+
+    def test_delete_user_sessions(self):
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        self.do_test_session('hamlet@zulip.com', lambda: delete_user_sessions(user_profile), True)
+        self.do_test_session('othello@zulip.com', lambda: delete_user_sessions(user_profile), False)
+
+    def test_delete_realm_user_sessions(self):
+        realm = get_realm('zulip')
+        self.do_test_session('hamlet@zulip.com', lambda: delete_realm_user_sessions(realm), True)
+        self.do_test_session('sipbtest@mit.edu', lambda: delete_realm_user_sessions(realm), False)
+
+    def test_delete_all_user_sessions(self):
+        self.do_test_session('hamlet@zulip.com', lambda: delete_all_user_sessions(), True)
+        self.do_test_session('sipbtest@mit.edu', lambda: delete_all_user_sessions(), True)
+
+    def test_delete_all_deactivated_user_sessions(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile_1 = get_user_profile_by_email('hamlet@zulip.com')
+        user_profile_1.is_active = False
+        self.client_post('/accounts/logout/')
+        delete_all_deactivated_user_sessions()
```

Comment:
Sorry I think I just forgot to include the check for the user's session. I had that line of code so that `delete_all_deactivated_user_sessions()` would work, without that `delete_all_deactivated_user_sessions()` wouldn't do anything. I think the reason is probably a check that the function performs before deleting a session.